pkg/controller/sidecar/sync.go

```diff
@@ -104,6 +104,36 @@ func (c *Controller) decommissionNode(ctx context.Context, svc *corev1.Service)
 	return nil
 }
 
+func (c *Controller) setHostIdAnnotation(ctx context.Context, svc *corev1.Service, hostId string) error {
+	svcCopy := svc.DeepCopy()
+	svcCopy.Annotations[naming.HostIdAnnotation] = hostId
+	_, err := c.kubeClient.CoreV1().Services(svcCopy.Namespace).Update(ctx, svcCopy, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Controller) syncHostIdAnnotation(ctx context.Context, svc *corev1.Service) error {
+	hostId, err := c.getHostId(ctx)
+	if err != nil {
+		return err
+	}
+
+	hostIdValue, hasHostIdAnnotation := svc.Annotations[naming.HostIdAnnotation]
+	if hasHostIdAnnotation && hostIdValue == hostId {
+		return nil
+	}
+
+	err = c.setHostIdAnnotation(ctx, svc, hostId)
```

Comment:
nit: pls inline (this is an opinionated function that could be mistaken for setting the annotation on the Service object instead of calling the API, the name is not the best fit)